src/coreclr/jit/flowgraph.cpp

```diff
@@ -926,6 +1174,13 @@ GenTreeCall* Compiler::fgGetStaticsCCtorHelper(CORINFO_CLASS_HANDLE cls, CorInfo
     GenTreeCall* result = gtNewHelperCallNode(helper, type, argList);
     result->gtFlags |= callFlags;
 
+    if (helper == CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE)
```

Comment:
You should be able to do this for the dynamic static helper too.

Reply:
Seems like you still have some work to do here?